stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/util/DownloadUtils.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2014-2023 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.util
+
+import android.Manifest
+import android.content.Context
+import android.os.Build
+import android.os.Environment
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import com.getstream.sdk.chat.utils.extensions.onPermissionRequested
+import com.getstream.sdk.chat.utils.extensions.wasPermissionRequested
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.offline.extensions.downloadAttachment
+import io.getstream.chat.android.uiutils.extension.openSystemSettings
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+internal fun attachmentDownloadState(): Pair<PermissionState, MutableState<Attachment?>> {
+    var writePermissionRequested by rememberSaveable { mutableStateOf(false) }
+    val writePermissionState = rememberPermissionState(
+        Manifest.permission.WRITE_EXTERNAL_STORAGE
+    ) {
+        writePermissionRequested = true
+    }
+
+    val downloadPayload = remember { mutableStateOf<Attachment?>(null) }
+
+    val context = LocalContext.current
+
+    LaunchedEffect(writePermissionState.status.isGranted) {
+        if (writePermissionState.status.isGranted) downloadPayload.value?.let {
```

Comment:
Small suggestion but maybe using brackets for the if statement would be an improvement in readability.